src/common/task.c

```diff
@@ -269,47 +281,58 @@ void print_task(task_spec *spec, UT_string *output) {
 
 /* TASK INSTANCES */
 
-struct task_instance_impl {
-  task_iid iid;
-  int32_t state;
+struct task_impl {
+  scheduling_state state;
   node_id node;
   task_spec spec;
 };
 
-task_instance *make_task_instance(task_iid task_iid,
-                                  task_spec *spec,
-                                  int32_t state,
-                                  node_id node) {
-  int64_t size = sizeof(task_instance) - sizeof(task_spec) + task_size(spec);
-  task_instance *result = malloc(size);
+bool node_ids_equal(node_id first_id, node_id second_id) {
+  return UNIQUE_ID_EQ(first_id, second_id) ? true : false;
```

Comment:
this is inconsistent with how `object_ids_equal` is defined (that one doesn't use the ternary operator). This way is probably better safer, although I'm not totally sure how the `bool` implementation here works.